keps/sig-cloud-provider/20191004-out-of-tree-credential-providers.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Out-of-Tree Credential Providers
+authors:
+  - "@mcrute"
+owning-sig: sig-cloud-provider
+participating-sigs:
+  - sig-node
+  - ug-aws
+  - ug-gcp
+  - ug-azure
+reviewers:
+  - "@andrewsykim"
+  - "@cheftako"
+  - "@nckturner"
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-10-04
+last-updated: 2019-10-06
+status: provisional
```

Comment:
Not against provisional but this indicate we agree on the problem but not the solution. If we do not merge it as implementable by the 15th, then work cannot begin on it.

Reply:
I had the date wrong as the 17th for the deadline here. In either case I'd like to drive some consensus on what the problem statement is and to that end I've removed most of the concrete proposal in favor of a few options to consider for the solution and so that we can focus on the problem statement.

Regardless of if this KEP gets merged, I'd like to start some prototyping so we can see these solutions in action and drive some consensus on the solution as well. From there maybe we can move this to implementable and merge it concurrently with the actual solution.